api/proto/info/info.proto

```diff
@@ -405,6 +405,27 @@ enum ZDeviceState {
     ZDEVICE_STATE_BOOTING = 5;
 }
 
+enum StorageStatus {
+  STORAGE_STATUS_UNSPECIFIED = 0;
+  STORAGE_STATUS_ONLINE = 1;   // The device or virtual device is in normal working order.
+  STORAGE_STATUS_DEGRADED = 2; // The virtual device has experienced a failure but can still function.
+  STORAGE_STATUS_FAULTED = 3;  // The device or virtual device is completely inaccessible.
+  STORAGE_STATUS_OFFLINE = 4;  // The device has been explicitly taken offline by the administrator.
+  STORAGE_STATUS_UNAVAIL = 5;  // The device or virtual device cannot be opened. In some cases, pools with UNAVAIL devices appear in DEGRADED mode.
+  STORAGE_STATUS_REMOVED = 6;  // The device was physically removed while the system was running.
+  STORAGE_STATUS_SUSPENDED = 7; // A pool that is waiting for device connectivity to be restored.
+}
+
+message StorageDiskState {
+  string disk_name = 1;
+  StorageStatus status = 2;            // In ZFS
+}
+
+message StorageState {
+  StorageStatus storage_state = 1;      // zfs volume status
```

Comment:
@petr-zededa Any specific reason we are not keeping pool state on top level of ```StorageInfo``` structure?

Reply:
Good point. It seems we can move content of `StorageState` to the top of `StorageInfo` and remove `StorageState` to reduce complexity. WDYT @vk-en ?